pkg/profileparser/profileparser.go

```diff
@@ -294,19 +293,19 @@ func ParseProfilesAndDo(contentDom *xmldom.Document, pb *cmpv1alpha1.ProfileBund
 	return nil
 }
 
-func parseProfileFromNode(profileRoot *xmldom.Node, pb *cmpv1alpha1.ProfileBundle, defType cmpv1alpha1.ComplianceScanType, defName, nonce string, action func(p *cmpv1alpha1.Profile) error) error {
-	profileObjs := profileRoot.Query("//Profile")
+func parseProfileFromNode(profileRoot *xmlquery.Node, pb *cmpv1alpha1.ProfileBundle, defType cmpv1alpha1.ComplianceScanType, defName, nonce string, action func(p *cmpv1alpha1.Profile) error) error {
+	profileObjs := xmlquery.Find(profileRoot, "//xccdf-1.2:Profile")
 	for _, profileObj := range profileObjs {
 
-		id := profileObj.GetAttributeValue("id")
+		id := profileObj.SelectAttr("id")
 		if id == "" {
 			return LogAndReturnError("no id in profile")
 		}
-		title := profileObj.FindOneByName("title")
+		title := profileObj.SelectElement("xccdf-1.2:title")
```

Comment:
nit - let's use a constant for the "xccdf-1.2:" prefix

Reply:
I wonder how the compiler would optimize this... would it do the sprintf operation on each instance? or does it optimize it to just set the resulting string?

Reply:
To my knowledge the '+' concatenation allocates a new string and combines the two, and performs worse as the strings become larger, so for short strings it would be quicker than an sprintf. But keeping it like it is now would be even better, so let's ignore this nit.